### PR TITLE
Fix/55 old media

### LINKED ISSUE #55: Upgrading from v0.12/lower to v0.13 breaks existing media
Despite making changes to support the new API, v0.13 uses a post meta value to store the source URL. This meta is not present in assets used in 0.12, resulting in broken images on sites that upgrade to the new version.

This means either a migration is required to use 0.13, or a fallback needs to be implemented, but neither are mentioned in the release notes.

Issue tracked down to changes in https://github.com/humanmade/asset-manager-framework/pull/35

### PR DESCRIPTION
**Pull request by tomjn**
A WIP attempt to fix or at least mitigate #55

fixes #55

### CODE REVIEW

**Comment by tomjn** [comment]
Noting that this currently has an infinite loop bug in the fix to `add_fallback_sizes`

**Comment by lebz** [comment]
The new changes solved the problem of empty images, and the PHP notice.

File: inc/namespace.php [hunk 1]
```diff
@@ -222,13 +222,22 @@ function replace_attached_file( $file, int $attachment_id ) : string {
...
 function replace_attachment_url( $url, int $attachment_id ) : string {
 	$attachment = get_post( $attachment_id );
 	if ( ! is_amf_asset( $attachment ) ) {
 		return $url ?: '';
 	}
 
-	return wp_unslash( get_post_meta( $attachment_id, '_amf_source_url', true ) ?: '' );
+	$meta_url = get_post_meta( $attachment_id, '_amf_source_url', true );
+
+	return wp_unslash(  $meta_url ?: $url );
```
Comment (svandragt) [inline inc/namespace.php:240]: I'd change the function name to `maybe_replace_attachment_url` and comment why there would be no attachment available, to improve expectations for developer use.

File: inc/namespace.php [hunk 2]
```diff
@@ -384,9 +393,12 @@ function add_fallback_sizes( array $metadata, int $attachment_id ) : array {
 		$metadata['sizes'] = [];
 	}
 
+	$source_url = get_post_meta( $attachment_id, '_amf_source_url', true )
```
Comment (svandragt) [inline inc/namespace.php:396]: I suggest to encapsulate getting the source url that future developers don't have to know  to check for a fallback. They can just pass the fallback as a function parameter to something like `get_source_url($attachment_id, $fallback_url)`.

Especially as this is done in multiple places and the next developer might make the same bug.

**Review by svandragt — DISMISSED** [review]
Fix seems good, but I think let's make it less prone to user error.

**Comment by tomjn** [comment]
@svandragt changes made

**Review by svandragt — APPROVED** [review]
Happy with this, nice work all round!

**Comment by roborourke** [comment]
Nice one @tomjn, released in 0.13.3

**Pull request merged**